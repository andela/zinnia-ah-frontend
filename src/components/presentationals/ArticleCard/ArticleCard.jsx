import React from 'react';
import PropTypes from 'prop-types';

import {
  DEFAULT_ARTICLE_PICTURE,
  DEFAULT_PROFILE_PICTURE,
} from '../../../config/config';

//style
import 'semantic-ui-css/semantic.min.css';
import './ArticleCard.scss';

import Title from '../Title/Title';
import UserThumbnail from '../UserThumbnail/UserThumbnail';

const Article = props => {
  const { article, user } = props;
  if (user) user.image = user.image || DEFAULT_PROFILE_PICTURE;
  return (
    <div className="article">
      <div className="thumbnail">
        <a href={article.slug}>
          <img src={article.imageThumbnail || DEFAULT_ARTICLE_PICTURE} />
        </a>
      </div>
      <div className="description">
        <a href={article.slug}>
          <Title content={article.title} className="ui header" />
        </a>
        <p>{article.description}</p>
        {user && (
          <div className="footer">
            <UserThumbnail
              name={`${user.firstName} ${user.lastName}`}
              info={`@${user.username}`}
              image={user.image}
              url={user.username}
            />
            <a href={article.slug} className="btn btn-white read-more">
              Read More
            </a>
          </div>
        )}
      </div>
    </div>
  );
};

Article.propTypes = {
  article: PropTypes.any.isRequired,
  user: PropTypes.any,
};

export default Article;
