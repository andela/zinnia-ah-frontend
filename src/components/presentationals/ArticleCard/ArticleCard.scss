.articles {
  display: flex;
  flex-wrap: wrap;
  margin: 2rem auto;
  width: 95%;

  .article {
    display: flex;
    margin: 1rem 0;
    min-height: 10rem;
    min-width: 35rem;
    transition: transform 13s ease-in-out;
    width: 50%;
  }

  .article .thumbnail {
    background: rgb(247, 247, 247);
    flex-basis: 35%;
    overflow: hidden;

    img {
      height: 100%;
      max-height: 13rem;
      transform: scale(1.12);
      transition: transform 666ms ease-in-out;
      width: 100%;

      &:hover {
        filter: drop-shadow(16px 16px 20px #ccc);
        transform: scale(1.21);
      }
    }
  }

  .article .description {
    flex-basis: 60%;
    padding: 15px 10px;

    .header {
      color: #000;
      font-family: Lora;
      font-size: 2rem;
      font-style: normal;
      font-weight: bold;
      line-height: 31px;
      overflow: hidden;
      padding-bottom: 3px;
      text-overflow: ellipsis;
      white-space: nowrap;
      width: 300px;
    }

    p {
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 4;
      color: #333;
      display: -webkit-box;
      font-family: Lato;
      font-size: 1.05rem;
      font-style: normal;
      font-weight: normal;
      height: 5rem;
      line-height: 17px;
      overflow: hidden;
    }

    .footer {
      align-items: stretch;
      display: flex;
      flex-direction: row;
      padding-top: 0;

      .read-more {
        flex-basis: 30%;
        padding-top: 10px;
        width: 82px;
      }
    }
  }
}
