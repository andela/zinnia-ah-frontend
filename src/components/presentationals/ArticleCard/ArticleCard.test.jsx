import React from 'react';
import { mount } from 'enzyme';
import checkPropTypes from 'check-prop-types';

import ArticleCard from './ArticleCard';

const articles = {
  author: {
    firstName: 'example',
    lastName: 'purpose',
    image: 'image_url.png',
    username: 'example1',
  },
  slug: 'article-card-title-1234',
  title: 'article card title',
  description: 'article card testing sample, layout',
};

describe('ArticleCard::: ', () => {
  it('should render ArticleCard correctly', () => {
    const component = mount(<ArticleCard article={articles} />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('should receive props', () => {
    const expectedProps = { article: { slug: 'some-random-title-slug' } };

    const propsError = checkPropTypes(
      ArticleCard.propTypes,
      expectedProps,
      'props',
      ArticleCard.article,
    );
    expect(propsError).toBeUndefined();
  });
});
