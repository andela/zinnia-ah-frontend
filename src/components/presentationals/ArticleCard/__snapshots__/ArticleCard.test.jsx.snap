// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard:::  should render ArticleCard correctly 1`] = `
<Article
  article={
    Object {
      "author": Object {
        "firstName": "example",
        "image": "image_url.png",
        "lastName": "purpose",
        "username": "example1",
      },
      "description": "article card testing sample, layout",
      "slug": "article-card-title-1234",
      "title": "article card title",
    }
  }
>
  <div
    className="article"
  >
    <div
      className="thumbnail"
    >
      <a
        href="article-card-title-1234"
      >
        <img
          src="https://alagirart.com/wp-content/uploads/2015/07/import_placeholder.png"
        />
      </a>
    </div>
    <div
      className="description"
    >
      <a
        href="article-card-title-1234"
      >
        <Title
          className="ui header"
          content="article card title"
        >
          <h1
            className="title ui header"
          >
            article card title
          </h1>
        </Title>
      </a>
      <p>
        article card testing sample, layout
      </p>
    </div>
  </div>
</Article>
`;
